src/server/connection.rs

```diff
@@ -1513,16 +1560,22 @@ impl Connection {
                 }
             });
             #[cfg(all(windows, feature = "flutter"))]
-            std::thread::spawn(|| {
+            std::thread::spawn(move || {
                 if crate::is_server() && !crate::check_process("--tray", false) {
-                    crate::platform::run_as_user(vec!["--tray"]).ok();
+                    crate::platform::run_as_user(vec!["--tray"], usid).ok();
                 }
             });
         }
     }
 
     async fn on_message(&mut self, msg: Message) -> bool {
         if let Some(message::Union::LoginRequest(lr)) = msg.union {
+            #[cfg(target_os = "windows")]
+            {
+                if !self.checked_multiple_session {
+                    self.user_session_id = Some(crate::platform::get_current_process_session_id());
+                }
+            }
```

Comment:
Still. 
![1707121490108](https://github.com/rustdesk/rustdesk/assets/14891774/12e3bc8a-2103-4f23-8d1f-ab6d06930101)

And line 1826~1830, could you explain why `self.user_session_id` is assigned in these two places?
```
                    let usid;
                    match self.lr.option.user_session.parse::<u32>() {
                        Ok(n) => usid = Some(n),
                        Err(..) => usid = None,
                    }
                    self.user_session_id = usid;
```
Please try remove field `user_session_id ` and test `crate::platform::run_as_user(args.clone(), Some(crate::platform::get_current_process_session_id()));`

Reply:
It is still happening. But I think we can ignore it because in my testing it only occurred once in a while when rapidly reconnecting to different user sessions, it doesn't occur if there's a slight delay between connections.
I still did `crate::platform::run_as_user(args.clone(), Some(crate::platform::get_current_process_session_id()));` because it is cleaner.

Reply:
The reason may be that cm does not exit so quickly. Please find the reason and a solution in order to avoid safety hazards.

Reply:
`get_current_process_session_id(); was giving incorrect id once in a while on repeatedly changing the sessions between connections`
Please dig it out, because `get_current_process_session_id` is used in `run_as_user` now.

Reply:
> get_current_process_session_id(); was giving incorrect id once in a while on repeatedly changing the sessions between connections

I think it's related to how windows handles processes, it might be due to some short-term cache.
https://www.geeksforgeeks.org/context-switch-in-operating-system/

It is not a problem after my last commit, since we are not using `get_current_process_session_id` in case of switching user sessions. I have tested thoroughly and got the correct session id every time.